pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -114,6 +114,12 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: TopologyManager %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	if _, err := ParseResourceList(kc.KubeReserved); err != nil {
```

Comment:
we should also validate that the input is restricted to the set the Kubelet currently supports for KubeReserved and SystemReserved: 
- KubeReserved: cpu, memory, local ephemeral storage for the root filesystem
- SystemReserved: cpu, memory

/cc @dashpole 
to make sure I'm up-to-date on the supported ranges (if not please update the doc comments in the types.go files).

Reply:
It looks like according to this [doc](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) both KubeReserved and SystemReserved have support for `cpu`, `memory`, `ephemeral-storage`, and `pid`. So we should validate against these tags?

Reply:
We should do the same validation that the kubelet command does.  In my view a good way would be to move the list of resources into `pkg/kubelet` so that both the command and `ValidateKubeletConfiguration` can use the same list.

Reply:
To clarify, what does this list of resources look like?

Reply:
Look at how parseResourceList works -- specifically, the switch/case.  You might also have a function `IsValidResourceName` that checks a resource name against the list.

Reply:
So basically, what I understand, is to create a list in `pkg/kubelet` with the resource name string contstants, and add a function `IsValidResourceName` to `ParseResourceList` ?

Reply:
Yes; make sure to follow the existing logic in `cmd/kubelet/app/server.go`:

```			if v1.ResourceName(k) != pidlimit.PIDs || utilfeature.DefaultFeatureGate.Enabled(features.SupportNodePidsLimit) {
```

Reply:
From the looks of it, the case statement

`case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs:`

  in `ParseResourceList` is already validating against the valid resource names.